FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.SchemaManagement/ContainerRegistry/ContainerRegistryTemplateProvider.cs

```diff
@@ -60,7 +61,15 @@ public class ContainerRegistryTemplateProvider : IContainerRegistryTemplateProvi
             try
             {
                 var provider = _templateCollectionProviderFactory.CreateTemplateCollectionProvider(schemaImageReference, accessToken);
-                return await provider.GetTemplateCollectionAsync(cancellationToken);
+
+                // "TemplateManagementException" is base exception for "ContainerRegistryAuthenticationException" and "ImageFetchException"
+                return await Policy
+                  .Handle<TemplateManagementException>()
+                  .RetryAsync(3, onRetry: (exception, retryCount) =>
```

Comment:
Shall we use exponential back-off for the retry logic?

Reply:
Added, thanks.